src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -228,7 +252,7 @@ public void disconnect() {
   }
 
   public boolean isConnected() {
-    return socket != null && socket.isBound() && !socket.isClosed() && socket.isConnected()
```

Comment:
IsBound does not say if isConnected.

If may return null for disconnected TCP connections and also may return false for connected Unix Socket connections.